### PR TITLE
feat(ramp): add webview debug by env vars

### PR DESCRIPTION
**Pull request by wachunei**
## **Description**
This PR enables WebView debugging in android when either `RAMP_DEV_BUILD` or `RAMP_INTERNAL_BUILD` environment variables are set.

## **Manual testing steps**

1. ~~Create a production build in bitrise with `RAMP_INTERNAL_BUILD` set to `true`~~
    - Done in https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/982a293a-a2c4-4c9c-ad2f-72532469e74d 
3. When using the generated APK staging Ramp API is be used and Checkout View WebView must be debuggable

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -210,7 +210,8 @@ android {
 
 	  buildTypes.each {
 		  it.buildConfigField 'String', 'foxCode', "\"$System.env.MM_FOX_CODE\""
-
+      it.buildConfigField 'String', 'IS_RAMP_UAT', "\"$System.env.RAMP_INTERNAL_BUILD\""
```
Comment (sethkfman) [inline android/app/build.gradle:213]: Add comments on what each of these configs do.

File: android/app/src/main/java/io/metamask/MainApplication.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ public void onCreate() {
 			e.printStackTrace();
 		}
 
-		if (BuildConfig.DEBUG) {
+		if (BuildConfig.DEBUG || BuildConfig.IS_RAMP_UAT || BuildConfig.IS_RAMP_DEV) {
```
Comment (sethkfman) [inline android/app/src/main/java/io/metamask/MainApplication.java:87]: Add comments where each of these configs are generated in gradle

File: android/app/build.gradle (hunk 1 above)
Comment (wachunei) [reply android/app/build.gradle:214]: added in 5ae17fd73d867f247c3ed21e85d41d42836d0624

File: android/app/src/main/java/io/metamask/MainApplication.java (hunk 2 above)
Comment (wachunei) [reply android/app/src/main/java/io/metamask/MainApplication.java:87]: added in https://github.com/MetaMask/metamask-mobile/commit/5ae17fd73d867f247c3ed21e85d41d42836d0624

**Comment by wachunei** [comment]
I confirm [this build](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/982a293a-a2c4-4c9c-ad2f-72532469e74d?tab=artifacts) with `RAMP_INTERNAL_BUILD` set to `true` has the debugging mode on 
image

**Comment by wachunei** [comment]
I also confirm [this build](https://app.bitrise.io/build/fee6a9cf-05f1-46a8-afc2-1d4918eec01e?tab=artifacts) without the env var shows the debugging inspector 🤔 
image

File: android/app/src/main/java/io/metamask/MainApplication.java [hunk 3]
```diff
@@ -84,7 +84,7 @@ public void onCreate() {
 			e.printStackTrace();
 		}
 
-		if (BuildConfig.DEBUG) {
+		if (BuildConfig.DEBUG || BuildConfig.IS_RAMP_UAT != null || BuildConfig.IS_RAMP_DEV != null) {
```
Comment (wachunei) [inline android/app/src/main/java/io/metamask/MainApplication.java:87]: This not correct. Apparently `BuildConfig.IS_RAMP_UAT != null` is true even when the env var is not present. 

Probably the default value is still a string so it should be compared to `"true"`.

File: android/app/src/main/java/io/metamask/MainApplication.java [hunk 4]
```diff
@@ -84,7 +84,7 @@ public void onCreate() {
 			e.printStackTrace();
 		}
 
-		if (BuildConfig.DEBUG) {
+		if (BuildConfig.DEBUG || BuildConfig.IS_RAMP_UAT.equals("true") || BuildConfig.IS_RAMP_DEV.equals("true")) {
```
Comment (sethkfman) [inline android/app/src/main/java/io/metamask/MainApplication.java:87]: Can you comment where these values are being set (e.g. build.gradle and injected at build time)?

Comment (wachunei) [reply android/app/src/main/java/io/metamask/MainApplication.java:90]: added in ae408da81151cdb44d9a5e20e4e537e2534ae082

Comment (wachunei) [reply android/app/src/main/java/io/metamask/MainApplication.java:90]: fixed a typo here: 7feaf781ae0a9d7e23a4886a4aff158eb5d4dbe0

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by wachunei** [comment]
Blocked by https://github.com/MetaMask/metamask-mobile/pull/7345, will merge after resolution

**Comment by wachunei** [comment]
Blocked by #7347, will merge after resolution

**Pull request merged**